packages/seal/src/key-server.ts

```diff
@@ -115,3 +120,18 @@ export async function verifyKeyServer(server: KeyServer, timeout: number): Promi
 	const fullMsg = new Uint8Array([...DST_POP, ...server.pk, ...fromHex(server.objectId)]);
 	return bls12_381.verifyShortSignature(fromBase64(serviceResponse.pop), fullMsg, server.pk);
 }
+
+/**
+ * Verify the key server version. Throws an `InvalidKeyServerError` if the version is not supported.
+ *
+ * @param response - The response from the key server.
+ */
+export function verifyKeyServerVersion(response: Response) {
+	const keyServerVersion = response.headers.get('X-KeyServer-Version');
+	if (
+		keyServerVersion == null ||
+		new Version(keyServerVersion).older_than(SERVER_VERSION_REQUIREMENT)
+	) {
+		throw new InvalidKeyServerError('Key server version not found');
```

Comment:
also in case of old version, maybe change to
```suggestion
		throw new InvalidKeyServerError('Key server version not supported');
```